src/core/cm/imagemanager/imagemanager.hpp

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (C) 2025 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_CORE_CM_IMAGEMANAGER_IMAGEMANAGER_HPP_
+#define AOS_CORE_CM_IMAGEMANAGER_IMAGEMANAGER_HPP_
+
+#include <core/cm/fileserver/fileserver.hpp>
+#include <core/cm/launcher/itf/imageinfoprovider.hpp>
+#include <core/cm/smcontroller/itf/updateimageprovider.hpp>
+#include <core/common/spaceallocator/spaceallocator.hpp>
+#include <core/common/tools/identifierpool.hpp>
+#include <core/common/tools/timer.hpp>
+#include <core/common/types/types.hpp>
+
+#include "itf/encoder.hpp"
+#include "itf/fileinfoprovider.hpp"
+#include "itf/imagedecrypter.hpp"
+#include "itf/imagemanager.hpp"
+#include "itf/statusnotifier.hpp"
+#include "itf/storage.hpp"
+
+namespace aos::cm::imagemanager {
+
+/** @addtogroup cm Communication Manager
+ *  @{
+ */
+
+/**
+ * Image manager configuration.
+ */
+struct Config {
```

Comment:
I guess we shall move Config to config.hpp.

Reply:
Done